src/_pytest/python_api.py

```diff
@@ -478,6 +478,14 @@ def approx(expected, rel=None, abs=None, nan_ok: bool = False) -> ApproxBase:
       special case that you explicitly specify an absolute tolerance but not a
       relative tolerance, only the absolute tolerance is considered.
 
+    .. note::
+
+        Although ``approx`` can handle numpy arrays, you may also want to check
+        `numpy.testing`__ module, which have many possibilities of comparing
+        floating point numbers, if you are comparing two numpy arrays of floats.
```

Comment:
```suggestion
        ``approx`` can handle numpy arrays, but we recommend the
        specialised test helpers in `numpy.testing`__ if you need
        support for comparisons, NaNs, or ULP-based tolerances.
```

This is pretty pedantic, sorry, but the Numpy helpers support non-floats too (e.g. integers).  I also think it's worth spelling out what `numpy.testing` does that `pytest.approx` can't.

Reply:
Good point, applied!

Waiting for final approval before hitting merge. 👍

Reply:
Seems reasonable to me!